### ISSUE TITLE
Make brewing stands capable of using custom items to make custom items

### ISSUE THREAD

**Issue by Goldmattress** [Civcraft/FactoryMod#108]
Make brewing stands capable of using custom items to make custom items while still acting as if they were default minecraft brewing stands.

**Comment by WildWeazel** [comment]
This is more of a Contraptions feature

**Comment by rourke750** [comment]
Its not something too difficult.  It can be added into contraptions after.
Is that being worked on?

On Wed, Jun 24, 2015 at 12:08 AM, Travis Christian <notifications@github.com

**Comment by Goldmattress** [comment]
No contraptions has not had work on it for 3 months, until someone picks it up I will not consider an option for near future development

**Comment by ttk2** [comment]
If we don't push it it will never happen and we will forever waste time hacking stuff into FM or humbug, pretty much most of the structure is there at this point, adding a feature like this to contraptions would take about as much time as adding it to FM, therefore we should add it to contraptions and even if it only ends up going a very small number of things at least it will be in use to be extended .

**Comment by Goldmattress** [comment]
Get a lead dev fot contraptions until then tiny hacks for fm are easier and
quicker
There is aldo no reason this cannot be re used for contraptions

<notifications@github.com> wrote:
>
> If we don't push it it will never happen and we will forever waste time
hacking stuff into FM or humbug, pretty much most of the structure is there
at this point, adding a feature like this to contraptions would take about
as much time as adding it to FM, therefore we should add it to contraptions
and even if it only ends up going a very small number of things at least it
will be in use to be extended .

**Comment by WildWeazel** [comment]
I've already forked Contraptions and started looking at the design. I'll focus on developing it next once this round of FM stuff stabilizes. The more things we cram into FM, the more complicated Contraptions will have to be to seamlessly support all the old features.

**Comment by ttk2** [comment]
exactly my point wild, unless its already coded I don't support new non-standard factory additions to FM

**Comment by MonsieurWTF** [comment]
> If we don't push it it will never happen and we will forever waste time hacking stuff into FM or humbug

Pretty much unanimous response to you wanting to push Contraptions is

> Get a lead dev for contraptions

Put someone in charge. That's literally all you need to actually get that going again. @gmlaxfanatic was the main dev and he stopped making updates two months ago. The plugin hasn't budged since two months ago. Coincidence? 

Get someone who knows how to go forward with Contraptions from your dev team and then get people to work with them on building the rest of its features. It would help if @gmlaxfanatic was still around, but you can't just be waiting on them and expect others to keep moving without coordination.

**Comment by erocs** [comment]
You're now in charge WTF. Finish it up.

**Comment by MonsieurWTF** [comment]
@erocs I said from the dev team, someone who knows how to actually program.

**Comment by gmlaxfanatic** [comment]
But you were the annoited one WTF!

**Comment by erocs** [comment]
Yes, you are now the anointed one. If you can't code it yourself, find
someone who can. FM has basically been declared dead so one way or another
Contraptions needs to move forward.

**Comment by ProgrammerDan** [comment]
@MonsieurWTF You've come a long way already, time to upgrade from YAML to Java. You can do it!

**Comment by WildWeazel** [comment]
uh guys https://github.com/Bergecraft/Contraptions

**Comment by erocs** [comment]
(Sssh, I knew about that. :P  )

Regardless, writing a rant to increase Ttk2's workload is fairly
non-productive. Picking up work that is sitting there and getting it done
is productive. Civcraft could use a couple more Project Managers, other
then the sole PM Ttk2, to get stuff like this done. As it is, ranting just
places more stress on him. I'm amazed at how well he handles his workload
plus negative feedback.

**Comment by MonsieurWTF** [comment]
@gmlaxfanatic Wait so are you done with Contraptions or just on hiatus from Github?

@ProgrammerDan Yeah LOL, sure. Reading a config and knowing how that works doesn't mean I could go into Java easily and understand all of those without actual education on programming. Best I could do is delegate and ask for focus on certain things. This is also feeling like group sarcasm >_>.

@erocs So are you guys gonna be able to work on it as well, then?

@WildWeazel Is anything going on with that? It's just a clean fork.

**Comment by ProgrammerDan** [comment]
Can't speak for the others but I'm genuine. Best way to learn a language is to dive into it.You'll stumble a lot, sure, but you've got some great resources around you *at this moment in time* who can review and recommend better courses. 

Something as critical as Contraptions might be intimidating for a first project, but fearlessness is an *absolutely minimum requirement* to be a developer. Fearful developers are terrible, I've interviewed many and fired some.

**Comment by erocs** [comment]
Myself? I have a list of things to do around here which doesn't involve the
open source plugins.

**Comment by WildWeazel** [comment]
I haven't added anything yet because as I said above I've been evaluating the design and focusing on this FM upgrade. Dan is right about learning but tbh I wouldn't trust a plugin of this complexity to a new programmer.

**Comment by rourke750** [comment]
Oh god the possible if statements.

**Comment by MonsieurWTF** [comment]
@erocs > writing a rant to increase Ttk2's workload is fairly
non-productive. Picking up work that is sitting there and getting it done
is productive. Civcraft could use a couple more Project Managers, other
then the sole PM Ttk2, to get stuff like this done

I understand, I'm not just trying to rant at him, but from this perspective it seems like he's just asking you guys to work on it without actually giving someone lead to direct where it goes.

@ProgrammerDan Alright, if you say so. I'll need some help, probably via PM or on here, towards where to put Contraptions and how I should get it started back up again.

And no, I'm not intimidated by working on Contrap. as the lead, I'm just saying that I have 0 experience in creating new code. If you're fine with having me in that position then I'll do it.

**Comment by erocs** [comment]
> from this perspective it seems like he's just asking you guys to work on
it without actually giving someone lead to direct where it goes.

Very true. he's still working from an older model of Civcraft development
where he had to beg and hope someone would pick up the work. It's possible
that there's enough devs now to assign work, but then he doesn't have the
time to best make those assignments. The efforts here have grown to a point
where he needs PMs who can keep in mind what needs each project currently
has.

As far as Contraptions, the PM for it doesn't even have to be able to code.
Just keep it moving along and resolve any controversial decisions.

**Comment by jjj5311** [comment]
The problem with any project manager besides ttk means they still have to
rely on ttk for the final say.

**Comment by ttk2** [comment]
That's that best part of all of this, Civcraft gets multi-level management!

**Comment by erocs** [comment]
Nah, you're now the user of Civcraft open source software or maybe the CEO.
If you don't clearly state your requirements up front, you'll have to live
without until the next version. We need to decouple you from development if
you want any time back in your schedule and if you want this to scale.

**Comment by ttk2** [comment]
probably, I also need to train good people to delegate too, half the issue is that when I delegate to people they often don't take it nearly as seriously as I would.
